compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -363,8 +363,35 @@ impl Value {
                 HirExpression::Literal(HirLiteral::Str(unwrap_rc(value)))
             }
             // Format strings are lowered as normal strings since they are already interpolated.
-            Value::FormatString(value, _) => {
-                HirExpression::Literal(HirLiteral::Str(unwrap_rc(value)))
+            // We need to mock a FmtStr literal that matches the type expected elsewhere in the program.
+            // If we do not do this, we may end up with mismatched types
+            Value::FormatString(value, fmt_str_type) => {
+                // We include a space here as otherwise the ABI decoder will expect a value that does not exist.
+                let placeholder =
+                    interner.push_expr(HirExpression::Literal(HirLiteral::Str(" ".to_owned())));
```

Comment:
Why a space `" "` and not an empty string `""`?

Reply:
It is due to the ABI decoder expecting a certain number of fields. But your comment made me realize this hack is not going to work in scenarios in types with differing field counts. Will push an update.